common/lib/util/eventemitter.js

```diff
@@ -57,7 +57,7 @@ var EventEmitter = (function() {
 			/* ... or we take event to be the actual event name and listener to be all */
 		}
 		var listeners, idx = -1;
-		if(event === null) {
+		if((event === null) || (event === undefined)) {
```

Comment:
The reason I implemented this is because `null` was being passed to this method, see https://github.com/ably/ably-js/blob/master/common/lib/client/realtimechannel.js#L226-L227.

If we are going to insist that it's indeed `undefined`, then I am happy to change all methods that call it to assume `undefined` too, and I would also like to put a guard on there to reject in valid values to avoid confusion i.e. if a user assumes calling `unsubscribe(null, listener)` will unsubscribe listeners for all events. 

Are you happy for me to try and track down all uses of `null` then where we expected `undefined` where they use the `EventEmitter`?